### PR TITLE
Update cyclegan.py for compatibility with Tensorflow 2.11

### PR DESCRIPTION
**Pull request by pjpratik**
This commit updates the code to ensure compatibility with Tensorflow 2.11.  These changes ensure that the model can be properly saved and that the optimizer works as intended with the updated TensorFlow version.

### CODE REVIEW

File: examples/generative/cyclegan.py [hunk 1]
```diff
@@ -600,18 +600,18 @@ def discriminator_loss_fn(real, fake):
...
+    disc_Y_optimizer=keras.optimizers.legacy.Adam(learning_rate=2e-4, beta_1=0.5),
     gen_loss_fn=generator_loss_fn,
     disc_loss_fn=discriminator_loss_fn,
 )
 # Callbacks
 plotter = GANMonitor()
 checkpoint_filepath = "./model_checkpoints/cyclegan_checkpoints.{epoch:03d}"
 model_checkpoint_callback = keras.callbacks.ModelCheckpoint(
-    filepath=checkpoint_filepath
+    filepath=checkpoint_filepath,save_weights_only=True
```
Comment (fchollet) [inline examples/generative/cyclegan.py:614]: Please use `black` on the code

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

What issue do you encounter with the new optimizers?

**Comment by pjpratik** [comment]
@fchollet The line number 645 `cycle_gan_model.load_weights(weight_file).expect_partial()` restores the checkpoint from a legacy Keras optimizer.

Therefore, while loading into 2.11v results in value error.

Thanks.

**Review by fchollet — APPROVED** [review]
Sounds good! Please update the ipynb and md files accordingly.

**Comment by pjpratik** [comment]
Updated ipynb and md files with the changes.

Thanks.

File: examples/generative/md/cyclegan.md [hunk 2]
```diff
@@ -616,18 +616,18 @@ cycle_gan_model = CycleGan(
...
+    disc_Y_optimizer=keras.optimizers.legacy.Adam(learning_rate=2e-4, beta_1=0.5),
     gen_loss_fn=generator_loss_fn,
     disc_loss_fn=discriminator_loss_fn,
 )
 # Callbacks
 plotter = GANMonitor()
 checkpoint_filepath = "./model_checkpoints/cyclegan_checkpoints.{epoch:03d}"
 model_checkpoint_callback = keras.callbacks.ModelCheckpoint(
-    filepath=checkpoint_filepath
+    filepath=checkpoint_filepath, save_weights_only = True
```
Comment (fchollet) [inline examples/generative/md/cyclegan.md:630]: Remove spaces here

Comment (pjpratik) [reply examples/generative/md/cyclegan.md]: Thanks for the suggestion. Updated the .md.

File: examples/generative/md/cyclegan.md [hunk 3]
```diff
@@ -616,18 +616,18 @@ cycle_gan_model = CycleGan(
...
+    disc_Y_optimizer=keras.optimizers.legacy.Adam(learning_rate=2e-4, beta_1=0.5),
     gen_loss_fn=generator_loss_fn,
     disc_loss_fn=discriminator_loss_fn,
 )
 # Callbacks
 plotter = GANMonitor()
 checkpoint_filepath = "./model_checkpoints/cyclegan_checkpoints.{epoch:03d}"
 model_checkpoint_callback = keras.callbacks.ModelCheckpoint(
-    filepath=checkpoint_filepath
+    filepath=checkpoint_filepath,save_weights_only = True
```
Comment (fchollet) [inline examples/generative/md/cyclegan.md:630]: So this line should `filepath=checkpoint_filepath, save_weights_only=True` (same as the other files)

Comment (pjpratik) [reply examples/generative/md/cyclegan.md]: Thanks for the suggestion. Updated the file accordingly.

**Review by fchollet — APPROVED** [review]
Thanks!

**Pull request merged**